internal/controller/dnsrecord_controller.go

```diff
@@ -371,9 +381,13 @@ func (r *DNSRecordReconciler) applyChanges(ctx context.Context, dnsRecord *v1alp
 		return noRequeueDuration, err
 	}
 
-	dnsRecord.Status.ValidFor = defaultRequeueTime.String()
-	dnsRecord.Status.QueuedAt = reconcileStart
 	if plan.Changes.HasChanges() {
+		logger.Info("Applying changes")
+		err = registry.ApplyChanges(ctx, plan.Changes)
+		if err != nil {
+			return noRequeueDuration, err
+		}
+
```

Comment:
If there is a situation when generation has changed and we failed to `.ApplyChanges()` we will not reset the counter and on the next run (if we succeed to apply the changes) the generation will be the same so the counter will not be reset even through the generation change. Should we move this block in the beginning on the logic then? :thinking: I might be missing something